### PR TITLE
Add settings page to GLPI sync

### PR DESCRIPTION
**Pull request by nicolas-brousse**

### CODE REVIEW

File: app/views/external_app_records/settings.html.erb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+<%= render Page::HeadingComponent.new(
+  title: t(".title"), breadcrumb_steps:, back_button_url: external_app_records_path
+) %>
+<div class="col-12 p-4 border-top">
+  <%= form_with(model: @settings, url: settings_external_app_records_path, method: :put, html: { class: "col-12 col-md-10 col-lg-8 mx-auto", role: "form" }) do |f| %>
+    <%= render FormErrorsComponent.new(@settings) %>
+
+    <%= render CardComponent.new(extra_classes: "bg-body-tertiary") do |card| %>
+      <% card.with_header do %>
+        <%= "Categories" %>
```
Comment (MBausson) [inline app/views/external_app_records/settings.html.erb:16]: Locale is missing here

File: app/views/external_app_records/settings.html.erb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+          <div class="form-check">
+            <%= b.check_box(class: "form-check-input") %>
+            <%= b.label(class: "form-check-label") %>
+          </div>
+        <% end %>
+      </fieldset>
+    <% end %>
+
+    <div class="col-12 py-4 mt-4 text-end sticky-bottom bg-body-tertiary border-top">
+      <%= f.submit(class: "btn btn-info") %>
```
Comment (MBausson) [inline app/views/external_app_records/settings.html.erb:32]: The button label is incoherent : it is currently "Create" ( Créer ). "Save" should be more appropriate

File: app/controllers/external_app_records_controller.rb [hunk 3]
```diff
@@ -9,6 +9,16 @@ def index
     @pagy, @external_app_records = pagy(@external_app_records)
   end
 
+  def settings
+    @settings = ExternalAppRecordSetting.new(settings_params)
+
+    return unless params[:commit]
```
Comment (MBausson) [inline app/controllers/external_app_records_controller.rb:15]: I don't understand the point of this line

Comment (nicolas-brousse) [reply app/controllers/external_app_records_controller.rb:15]: `#settings` action is used for both print and apply the form. This is the name of the submit button, and so it let us know the user as submit the form

File: app/views/external_app_records/settings.html.erb (hunk 2 above)
Comment (nicolas-brousse) [reply app/views/external_app_records/settings.html.erb]: Fixed

File: app/views/external_app_records/settings.html.erb (hunk 1 above)
Comment (nicolas-brousse) [reply app/views/external_app_records/settings.html.erb]: Fixed

**Review by MBausson — APPROVED** [review]

**Review by B-Rass — APPROVED** [review]

**Pull request merged**